### PR TITLE
Distance eval, new experiments, and some clean up

### PR DESCRIPTION
**Pull request by colbybanbury**

### CODE REVIEW

File: wake_vision_loader.py [hunk 1]
```diff
@@ -337,44 +337,66 @@ def get_wake_vision(cfg=default_cfg, batch_size=None):
...
+        num_samples = cfg.COUNT_PERSON_SAMPLES_TEST
+    elif split == "validation":
+        num_samples = cfg.COUNT_PERSON_SAMPLES_VAL
+    else:
+        raise ValueError("split must be 'test' or 'validation'")
+    
+    wv_ds = open_images_to_wv(ds, num_samples, cfg=cfg)
+
+    # Create finer grained evaluation sets before preprocessing the dataset.
+    miaps = {
```
Comment (colbybanbury) [inline wake_vision_loader.py:359]: add no_person set to MIAPS

File: distance_val.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+# has been configured. The backend cannot be changed once the
+# package is imported.
+import keras
+
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+from wake_vision_loader import get_distance_eval
+
+def distance_val(model_cfg):
```
Comment (Ekhao) [inline distance_val.py:19]: Should we have a similar file for all the other validation and test sets?

File: distance_val.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        f"\n\tNo Person: {no_person_score[1]}")
+    
+    print(result)
+
+    return result
+
+if __name__ == "__main__":
+    # model_yaml = "gs://wake-vision-storage/saved_models/wv_large2023_12_28-06_14_17_PM/config.yaml"
+    # model_yaml = "gs://wake-vision-storage/saved_models/vww_large2023_12_28-06_14_17_PM/config.yaml"
+    # model_yaml = "gs://wake-vision-storage/saved_models/vww_small2023_12_28-05_08_30_AM/config.yaml"
```
Comment (Ekhao) [inline distance_val.py:47]:
```suggestion
```

File: experiment_config.py [hunk 4]
```diff
@@ -28,25 +28,30 @@ def get_cfg(experiment_name=None):
     cfg.SAVE_DIR = f"{cfg.BUCKET_NAME}saved_models/{cfg.EXPERIMENT_NAME}/"
     cfg.SAVE_FILE = cfg.SAVE_DIR + f"{cfg.MODEL_NAME}.keras"
 
-    cfg.COUNT_PERSON_SAMPLES_TRAIN = 844965  # Number of person samples in the WV train dataset. The number of non-person samples are 898077. We will use this number to balance the dataset.
-    cfg.COUNT_PERSON_SAMPLES_VAL = 9973  # There are 31647 non-person samples.
-    cfg.COUNT_PERSON_SAMPLES_TEST = 30226  # There are 95210 non-person samples. The distribution of persons in both the Val and Test set is close to 24% (Val:23.96) (Test:24.09) so we may not need to reduce the size of these.
-    cfg.MIN_IMAGE_LEVEL_CONFIDENCE = 5  # Minimum confidence level for image-level labels to be included in the dataset. Only used for the wake vision dataset. If 0 then even negatively human verified labels are included.
+
+    #TODO recalculate these numbers
```
Comment (Ekhao) [inline experiment_config.py:32]: We can do this before merging into main

File: experiment_config.py [hunk 5]
```diff
@@ -28,25 +28,30 @@ def get_cfg(experiment_name=None):
...
+    if cfg.LABEL_TYPE == "image":
+        cfg.COUNT_PERSON_SAMPLES_TRAIN = 3238953
+        cfg.COUNT_PERSON_SAMPLES_VAL = 19311
+        cfg.COUNT_PERSON_SAMPLES_TEST = 58288
+    cfg.COUNT_PERSON_SAMPLES_TRAIN = 675411#844965  # Number of person samples in the WV train dataset. The number of non-person samples are 898077. We will use this number to balance the dataset.
```
Comment (Ekhao) [inline experiment_config.py:37]: Seems like we need an elif here?

File: simple_scaling_experiment.py [hunk 6]
```diff
@@ -2,46 +2,51 @@
...
-vww_large_config = get_cfg("vww_large")
-vww_large_config.TARGET_DS = "vww"
-vww_large_config.EXPERIMENT_NAME = "vww_large"
-vww_large_config.MODEL_NAME = "vww_large_mobilenetv1"
-vww_large_config.SAVE_FILE = vww_large_config.SAVE_DIR + f"{vww_large_config.MODEL_NAME}.keras"
-vww_large_config.MODEL_SIZE = 1.0
-
-
-wv_large_config = get_cfg("wv_large")
+# wv_small_config = get_cfg("wv_small")
```
Comment (Ekhao) [inline simple_scaling_experiment.py:5]: Should everything in this file be commented out?

File: wake_vision_loader.py [hunk 7]
```diff
@@ -337,44 +337,66 @@ def get_wake_vision(cfg=default_cfg, batch_size=None):
...
-    for key, value in miaps_validation.items():
-        miaps_validation[key] = preprocessing(value, batch_size, cfg=cfg)
+    for key, value in miaps.items():
+        miaps[key] = preprocessing(value, batch_size, cfg=cfg)
+
+    return miaps
+
+
+#Distance Eval
+def get_distance_eval(cfg=default_cfg, split="test"):
```
Comment (Ekhao) [inline wake_vision_loader.py:376]: Should we make it possible to specify a batch size?

File: distance_val.py (hunk 2 above)
Comment (colbybanbury) [reply distance_val.py:19]: I think having a file to run each eval on a trained model is useful

Comment (Ekhao) [reply distance_val.py:19]: I'm on it atm :)

**Pull request merged**